code/__HELPERS/lists.dm

```diff
@@ -10,6 +10,16 @@
  */
 
 //Returns a list in plain english as a string
+
+#define LAZYINITLIST(L) if (!L) L = list()
+#define ISREALLIST(L) (L && istype(L,/list) && L.len)
+#define UNSETEMPTY(L) if(!L.len) L = null
+#define LAZYLEN(L) ( L ? L.len : 0 )
+#define LAZYREMOVE(L, I) if(L) { L -= I; if(!L.len) { L = null; } }
+#define LAZYADD(L, I) if(!L) { L = list(); } L += I;
+#define LAZYACCESS(L, I) (L ? (isnum(I) ? (I > 0 && I <= L.len ? L[I] : null) : L[I]) : null)
+#define LAZYLEN(L) length(L)
```

Comment:
> blame them.

Don't bring a pig into the house and then blame the owner of the pig when the pig ruins everything.